common/data_structure.go

```diff
@@ -2,4 +2,54 @@ package common
 
 type FileAttributes struct {
 	Size uint
-}
\ No newline at end of file
+}
+
+type BlockMetadata struct {
+	Size uint
+}
+
+type BlockReport struct {
+	Endpoint      string
+	BlockMetadata map[string]BlockMetadata
+}
+
+type Heartbeat struct {
+	Endpoint string
+}
+
+// Request to read a file with the specified name and block range [BeginBlock, EndBlock)
+type ReadFileRequest struct {
+	FileName   string
+	BeginBlock uint
+	EndBlock   uint
+}
+
+type ReadFileResponse struct {
+	Succeeded     bool
```

Comment:
Why do we need this field? The error can be indicated in the error of the RPC call.

Reply:
In this way, you can get rid of this data structure.

Reply:
Well, for the ease of adding new functionality you may keep this struct even though it's currently just a wrapper.